### ISSUE TITLE
contract scripts review

### ISSUE THREAD

**Issue by Taombawkry** [bicos-io01/Revenue-Sharing-Source#69]
### Review Request: Smart Contract Interaction Scripts

I am requesting a review of specific scripts in our project that interact with smart contracts. The review should focus on identifying areas needing optimization or correction in preparation for demo next month. Below are the scripts in question:

1. **Exhibit Deployment API**  
   - **Location**: `packages/dapp/src/app/api/deployExhibit`

2. **Event Escrow Distribute Client Component** 
   - **Location**: `packages/dapp/src/functionality/eventEscrowComponent.tsx`

3. **Museum - Client Component for Ticket Purchase** 
   - **Location**: `packages/dapp/src/functionality/ticketpurchasecomponent.tsx`
   
4. **ArtifactNFT Deployment & Minting**
   -  **Location**: `packages/contracts/scripts/deploycall.ts`

**Request**: Please provide  notes on areas of improvement, potential optimizations, and any corrections needed in the implementation of these scripts post review. Insights on best practices for smart contract interaction and front-end integration would be highly beneficial.

**Comment by daodesigner** [comment]
Quick notes:
- packages/dapp/src/app/api/deployExhibit/route.ts
Deploys from the Event organizer service so will get picked up by the subgraph. Try not to hardcode the gaslimit in prod. need to test this route is specially locked down as it can use the DEV_P_K for gas (superadmin or rate limit?).
- packages/dapp/src/functionality/eventEscrowComponent.tsx
Need to actually test with the session API auth but good to see the session check. useClient, also note that this needs to be used inside of seomthing wrapped in ApolloWrapper (right now all the (UserPages) are in the layout.
- packages/dapp/src/functionality/ticketpurchasecomponent.tsx
contract call looks good, might want to drop the number of blocks to wait down from 6 to 2 for better UX. useClient, if you want to use react lifecycle hooks.
- packages/contracts/scripts/deploycall.ts
should call .deployArtifactNFT on the organizerServiceContract passing the same parameters instead of deploying from ethers contract Factory. Doing it this way allows the subgraph tot track the artifact deployments. We should agree on the address for the organizerService to use on each network. 

should I open a branch with the suggested changes?

**Comment by Taombawkry** [comment]
That would be great, I'll be able to compare changes from that branch and make notes.
Especially on the last point.

Also on auth haven't yet protected the routes(it has been set up), added it proactively as users will need to be validated in order to access the fund distribution dashboard section.

On the deployExhibit, hardcoded the gas limit because I kept getting thrown the 'Unable to estimate gas' message. Couldn't find a better alternative solution to test so I hardcoded.
